### ISSUE TITLE
Problems when trying to debug existing xdebug instance

### ISSUE THREAD

**Issue by odahcam** [xdebug/vscode-php-debug#599]
PHP version: ^7
Xdebug version: ^2

Why can't this extension attach to an existing running process?

I have a docker container running xdebug with the 9000 port exposed and I wanted to simply attach my vs code to this port and debug the existing process.

Could this be a feature request?

**Comment by odahcam** [comment]
Launch vs. Attach by VSCode docs: https://code.visualstudio.com/docs/editor/debugging#_launch-versus-attach-configurations

**Comment by odahcam** [comment]
I only studied the code for a few seconds, but it seems that if we skip line 362:
https://github.com/xdebug/vscode-php-debug/blob/f29431ef39c06e521f23b4ad28735c21483e58fe/src/phpDebug.ts#L362
We would be creating an "attach" style process. Wouldn't it be possible?

**Comment by zobo** [comment]
Hi @odahcam.

Attaching the debugger to a PHP process is not possible, this is not how PHP/Xdebug work. The PHP process connects back to the IDE, so it either has to connect on startup or on error. It's described a bit more here: https://xdebug.org/docs/step_debug#configure

It's also currently not possible to "pause" a running PHP process even if the debugger is attached.

What's your use-case, maybe I can give a better suggestion depending on that.

Best!

**Comment by odahcam** [comment]
I'm sorry for such an inconvenient issue, honestly. I also spent all night studying and understanding how Xdebug and php-debug works.

Now I know that:

1. Xdebug PHP extension is the one who collects info and actually sends it to a server.
2. My machine thru the php-debug VSCode extension runs a web socket server (remote debugger) on port 9003 which implements the DGB protocol.
3. The extension running the server also connects it's info with VSCode's debugger.

But I still have a problem, my Xdebug can send information to my VSCode, somehow it seems to don't reach the remote server due to an error: 

[**IT SIMPLY WORKED ONE TIME AND STOPPED AGAIN** but let's complete the thoughts]

```sh
php_1 | [15-Jun-2021 13:36:28] WARNING: [pool www] child 8 said into stderr: "NOTICE: PHP message: Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-("

nginx_1 | PHP message: Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(" while reading upstream, client: 172.18.0.1, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://172.18.0.2:9000", host: "localhost:7100"
```

My config was: 

`launch.json`:

<img width="617" alt="image" src="https://user-images.githubusercontent.com/3942006/122063194-6e23f100-cdc6-11eb-93ae-2bf7a1497501.png">

`Dockerfile`:

<img width="850" alt="image" src="https://user-images.githubusercontent.com/3942006/122063319-898efc00-cdc6-11eb-8058-abab2abdd0e4.png">

`docker-compose.yml`:

<img width="754" alt="image" src="https://user-images.githubusercontent.com/3942006/122063398-9ca1cc00-cdc6-11eb-97d4-0e94c3109518.png">


And the craziest thing just happened: I was going to record my screen to show you that when I start the "Listen for XDebug in Docker" my VSCode accent color stills blue and it seems to not connect to the debugger, but **while recording the screen it actually works** and the VSCode gets orange and works! WTF huh?! I tested many times and I still have no f* idea on how or why it works, not recording screen -> not works, recording screen -> works. I never seen anything like this before, it doesn't makes any sense for me. 😨

**Comment by odahcam** [comment]
I'm sorry for such an inconvenient issue, honestly. I also spent all night studying and understanding how Xdebug and php-debug works.

Now I know that:

1. Xdebug PHP extension is the one who collects info and actually sends it to a server.
2. My machine thru the php-debug VSCode extension runs a web socket server (remote debugger) on port 9003 which implements the DGB protocol.
3. The extension running the server also connects it's info with VSCode's debugger.

But I still have a problem, my Xdebug can send information to my VSCode, somehow it seems to don't reach the remote server due to an error: 

[**I JUST SOLVED IT** but let's complete the thoughts]

```sh
php_1 | [15-Jun-2021 13:36:28] WARNING: [pool www] child 8 said into stderr: "NOTICE: PHP message: Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-("

nginx_1 | PHP message: Xdebug: [Step Debug] Could not connect to debugging client. Tried: host.docker.internal:9003 (through xdebug.client_host/xdebug.client_port) :-(" while reading upstream, client: 172.18.0.1, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://172.18.0.2:9000", host: "localhost:7100"
```

My config was: 

`launch.json`:

<img width="617" alt="image" src="https://user-images.githubusercontent.com/3942006/122063194-6e23f100-cdc6-11eb-93ae-2bf7a1497501.png">

`Dockerfile`:

<img width="850" alt="image" src="https://user-images.githubusercontent.com/3942006/122063319-898efc00-cdc6-11eb-8058-abab2abdd0e4.png">

`docker-compose.yml`:

<img width="754" alt="image" src="https://user-images.githubusercontent.com/3942006/122063398-9ca1cc00-cdc6-11eb-97d4-0e94c3109518.png">


---

I said I solved the problem, well what I did:

- Changed from port 9000 to 9003, it was correctly set in all places and probably didn't have anything to do with the fix.
- Removed `xdebug.discover_client_host=1` setting from Dockerfile.
- Stopped using the mac's Touch Bar for running the debugger server since it was starting the wrong config.
